chain/chaintest/action_test_helpers.go

```diff
@@ -81,3 +81,41 @@ func (test *ActionTest) Run(ctx context.Context, t *testing.T) {
 		}
 	})
 }
+
+// ActionBenchmark is paramatized benchmark. It calls Execute on the action with the passed parameters
+// and checks that all assertions pass. To avoid using shared state between runs, a new
+// state is created for each iteration using the provided `CreateState` function.
+type ActionBenchmark struct {
+	Name   string
+	Action chain.Action
+
+	Rules       chain.Rules
+	CreateState func() state.Mutable
+	Timestamp   int64
+	Actor       codec.Address
+	ActionID    ids.ID
+
+	ExpectedOutputs [][]byte
+	ExpectedErr     error
+
+	Assertion func(context.Context, *testing.B, state.Mutable)
+}
+
+// Run executes the [ActionBenchmark] and make sure all the benchmark assertions pass.
+func (test *ActionBenchmark) Run(ctx context.Context, b *testing.B) {
+	require := require.New(b)
+
+	for i := 0; i < b.N; i++ {
+		b.StopTimer()
+		state := test.CreateState()
+		b.StartTimer()
```

Comment:
To avoid doing this in the loop, could we create a slice of `b.N` states ahead of time?

We're going to be a little bit prone to issues like this: https://github.com/golang/go/issues/20875